src/extensions/scratch3_boost/index.js

```diff
@@ -0,0 +1,1963 @@
+ * @readonly
+ * @enum {number}
+ */
+const BoostPort = {
+    A: 55,
+    B: 56,
+    C: 1,
+    D: 2
+};
+
+/**
+ * Enum for indexed colors read by the Boost Vision Sensor
+ * @readonly
+ * @enum {number}
+ */
+const BoostColor = {
+    NONE: 255,
+    RED: 9,
+    BLUE: 3,
+    GREEN: 5,
+    YELLOW: 7,
+    WHITE: 10,
+    BLACK: 0
+};
+
+/**
+ * Enum for Message Types
+ * @readonly
+ * @enum {number}
+ */
+const BoostMessage = {
+    HUB_PROPERTIES: 0x01,
+    HUB_ACTIONS: 0x02,
+    HUB_ALERTS: 0x03,
+    HUB_ATTACHED_IO: 0x04,
+    ERROR: 0x05,
+    PORT_INPUT_FORMAT_SETUP_SINGLE: 0x41,
+    PORT_INPUT_FORMAT_SETUP_COMBINED: 0x42,
+    PORT_INFORMATION: 0x43,
+    PORT_MODEINFORMATION: 0x44,
+    PORT_VALUE: 0x45,
+    PORT_VALUE_COMBINED: 0x46,
+    PORT_INPUT_FORMAT: 0x47,
+    PORT_INPUT_FORMAT_COMBINED: 0x48,
+    OUTPUT: 0x81,
+    PORT_FEEDBACK: 0x82
+};
+
+/**
+ * Enum for Motor Subcommands (for 0x81)
+ * @readonly
+ * @enum {number}
+ */
+const BoostOutputSubCommand = {
+    START_POWER_PAIR: 0x02,
+    SET_ACC_TIME: 0x05,
+    SET_DEC_TIME: 0x06,
+    START_SPEED: 0x07,
+    START_SPEED_PAIR: 0x08,
+    START_SPEED_FOR_TIME: 0x09,
+    START_SPEED_FOR_TIME_PAIR: 0x0A,
+    START_SPEED_FOR_DEGREES: 0x0B,
+    START_SPEED_FOR_DEGREES_PAIR: 0x0C,
+    GO_TO_ABS_POSITION: 0x0D,
+    GO_TO_ABS_POSITION_PAIR: 0x0E,
+    PRESET_ENCODER: 0x14,
+    WRITE_DIRECT_MODE_DATA: 0x51
+};
+
+/**
+ * Enum for Startup/Completion information for an output command.
+ * Startup and completion bytes must be OR'ed to be combined to a single byte.
+ * @readonly
+ * @enum {number}
+ */
+const BoostOutputExecution = {
+    // Startup information
+    BUFFER_IF_NECESSARY: 0x00,
+    EXECUTE_IMMEDIATELY: 0x10,
+    // Completion information
+    NO_ACTION: 0x00,
+    COMMAND_FEEDBACK: 0x01
+};
+
+/**
+ * Enum for Boost Motor end states
+ * @readonly
+ * @enum {number}
+ */
+const BoostMotorEndState = {
+    FLOAT: 0,
+    HOLD: 126,
+    BRAKE: 127
+};
+
+/**
+ * Enum for Boost Motor acceleration/deceleration profiles
+ * @readyonly
+ * @enum {number}
+ */
+const BoostMotorProfile = {
+    DO_NOT_USE: 0x00,
+    ACCELERATION: 0x01,
+    DECELERATION: 0x02
+};
+
+/**
+ * Enum for when Boost IO's are attached/detached
+ * @readonly
+ * @enum {number}
+ */
+const BoostIOEvent = {
+    ATTACHED: 0x01,
+    DETACHED: 0x00,
+    ATTACHED_VIRTUAL: 0x02
+};
+
+/**
+ * Enum for selected sensor modes.
+ * @enum {number}
+ */
+const BoostMode = {
+    TILT: 0, // angle (pitch/yaw)
+    LED: 1, // Set LED to accept RGB values
+    COLOR: 0, // Read indexed colors from Vision Sensor
+    MOTOR_SENSOR: 2, // Set motors to report their position
+    UNKNOWN: 0 // Anything else will use the default mode (mode 0)
+};
+
+/**
+ * Helper function for converting a JavaScript number to an INT32-number
+ * @param {number} number - a number
+ * @return {array} - a 4-byte array of Int8-values representing an INT32-number
+ */
+const numberToInt32Array = function (number) {
```

Comment:
Good point! If we want to do that though, I think we'd have to make them more generic, since this one is hardcoded to return "little endian"-format numbers.